src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3374,7 +3378,9 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
                   // the overloaded type into a single function type from which `typedFunction`
                   // can derive the argument type for `x` in the function literal above
                   val argPt =
-                    if (argPtAlts.nonEmpty && treeInfo.isFunctionMissingParamType(tree)) functionProto(argPtAlts)
+                    if (argPtAlts.isEmpty) WildcardType
+                    else if (treeInfo.isFunctionMissingParamType(tree)) functionProto(argPtAlts)
+                    else if (treeInfo.isPartialFunctionMissingParamType(tree)) partialFunctionProto(argPtAlts)
```

Comment:
My first impulse was to write "this works as intended and specified in https://github.com/scala/scala/pull/5698" but the spec change doesn't actually mandate it and it would be more efficient to create a `Function1` in cases where that's sufficient.